integrationtest-service-framework/src/main/java/org/hypertrace/core/common/grpc/client/TenantIdClientInterceptor.java

```diff
@@ -0,0 +1,49 @@
+package org.hypertrace.core.common.grpc.client;
+
+import static io.grpc.Metadata.ASCII_STRING_MARSHALLER;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingClientCall;
+import io.grpc.ForwardingClientCallListener;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+
+/**
+ * Implementation of {@link ClientInterceptor} which sets the tenantId in RequestContext
+ * in every request. Only used for testing because in real setup, envoy proxy does this logic.
+ */
+public class TenantIdClientInterceptor implements ClientInterceptor {
```

Comment:
Can you use grpc client utils instead of this?

Reply:
I'm guessing for some reason this got left out in the platform-gateway/integrationtest-service-framework and we we are just copying it here.
Once this is done, I'm assuming we'll the module from platform-gateway?

Reply:
My question is why we need this class at all. Why can't we just use the grpc client utils like we do in the services internally?

Reply:
Yes, it can. It's less convenient because client-utils is static and it needs to keep passing tenant-id for every call made in integration test but, it can be solved with creating a wrapper client of service under test.